src/Execution/lbsRuntime.py

```diff
@@ -216,6 +217,27 @@ def information_stage(self, n_rounds, f):
         n_u, v_min, v_ave, v_max, _, _, _, _ = compute_function_statistics(viewers_counts.values(), lambda x: x)
         self.lgr.info(f"Reporting viewers counts (min:{v_min}, mean: {v_ave:.3g} max: {v_max}) to {n_u} loaded ranks")
 
+    def recursive_extended_search(self, pick_list, object_list, c_fct):
+        """Recursively extend search to other objects
+        """
+
+        # Terminate negatively when pick list is empty
+        if not pick_list:
+            return False
+
+        # Pick one object and move it from one list to the other
+        o = random.choice(pick_list)
+        pick_list.remove(o)
+        object_list.append(o)
+
+        # Decide whether criterion allows transfer
+        if c_fct(object_list) < 0.:
+            # Transfer is not possible, recurse further
+            self.recursive_extended_search(pick_list, object_list, c_fct)
```

Comment:
Can we make this an configuration option so we can explore both ways easily?

Reply:
Yes, creating an issue for that. Thanks.